src/directives/offense/poisonRoom.ts

```diff
@@ -1,169 +1,188 @@
+	static secondaryColor = COLOR_BROWN;
+	static requiredRCL = 4;
+	walkableSourcePosisions: RoomPosition[];
+	walkableControllerPosisions: RoomPosition[];
+	overlords: {
+		claim: ClaimingOverlord;
+		roomPoisoner: RoomPoisonerOverlord;
+	};
+	constructor(flag: Flag) {
+		super(flag, colony => colony.level >= DirectivePoisonRoom.requiredRCL);
+		// Remove if misplaced
+		if (Cartographer.roomType(this.pos.roomName) != ROOMTYPE_CONTROLLER) {
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} is not a controller room; ` +
+						`removing directive!`);
+			this.remove(true);
+		}
+		// failsafe - if already owned and controller level > 2, remove flag
+		if(this.room && this.room.controller && this.room.controller.my && this.room.controller.level > 2) {
+			this.remove(true);
+		}
+		// Remove if owned by other player
+		if (this.room && !!this.room.owner && this.room.owner != MY_USERNAME) {
+			log.notify(`Removing poisonRoom directive in ${this.pos.roomName}: room already owned by another player.`);
+			this.remove(true);
+		}
+		// Remove if not enough GCL
+		if(getAllColonies().length == Game.gcl.level) {
+			log.warning(`${this.print}: ${printRoomName(this.pos.roomName)} not enough GCL; ` +
+						`for poison room, removing directive!`);
+		}
+	}
+	spawnMoarOverlords() {
+		this.overlords.claim = new ClaimingOverlord(this);
+		this.overlords.roomPoisoner = new RoomPoisonerOverlord(this);
+	}
+	init() {
+		this.alert(`Poisoning Room ${this.pos.roomName}`);
+
+		if(!(this.room && this.room.controller)) {
+			return;
+		}
+
+		// suspend room immediately once claimed to avoid default room actions
+		if(this.room.controller.level == 1) {
+			this.suspendColony(this.room.name);
+		}
+
+		if (Game.time % RUN_TIMER == 0 || this.room.constructionSites.length == 0) {
+			// capture variables
+			this.walkableSourcePosisions	 = _.filter(_.flatten(_.map(this.room.sources,
+													s => s.pos.neighbors)),pos => pos.isWalkable(true));
+			this.walkableControllerPosisions =  _.filter(this.room.controller!.pos.neighbors,
+													pos => pos.isWalkable(true));
+			
+			// reverse reservation if needed
+			if(this.room.controller.reservation && this.room.controller.reservation.ticksToEnd > 500) {
+				DirectiveControllerAttack.createIfNotPresent(this.room.controller.pos,'room');
+			}
+
+			// send fighters if needed
+			if(this.room && this.room.playerHostiles.length > 0) {
+				DirectiveOutpostDefense.createIfNotPresent(this.room.controller.pos,'room');
+			}
+		}
+	}
+	private suspendColony(roomName: string): string {
+		if (Overmind.colonies[roomName]) {
+			const colonyMemory = Memory.colonies[roomName] as ColonyMemory | undefined;
+			if (colonyMemory) {
+				colonyMemory.suspend = true;
+				Overmind.shouldBuild = true;
+				return `Colony ${roomName} suspended.`;
+			} else {
+				return `No colony memory for ${roomName}!`;
+			}
+		} else {
+			return `Colony ${roomName} is not a valid colony!`;
+		}
+	}
```

Comment:
I believe there exists a function to do this already. If this is doing something special (perhaps the build part), add more comments for it.

Reply:
there is a function to do this from the command line but this is not the correct way to do this, as calling it in init() will still result in the colony being instantiated if you are on a build tick.

Reply:
How about you do the following:
- Change the primary color to COLOR_PURPLE. Anything that involves claiming a room or that involves modifications to how the `Overmind` object treats colonies should be a colonizing directive with purple primary color.
- I will add a specific check for this directive and for `DirectiveClearRoom` which prevents the colony from being instantiated
- Remove all the suspension logic in this directive.

Reply:
I copied this from console.ts.
OK, so I will remove it and once merged, Muon can suspend it from somewhere else.